### PR TITLE
Fix WorkspaceService to be able to share workspaces

### PR DESCRIPTION
**Pull request by sleshchenko**
@skabashnyuk  Can you please review my changes?
@evoevodin Can you please take a look changes in WorkspaceManager?

### CODE REVIEW

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/WorkspaceDao.java [hunk 1]
```diff
@@ -133,13 +133,13 @@
     List<WorkspaceImpl> getByNamespace(String namespace) throws ServerException;
 
     /**
-     * Gets list of workspaces where user is worker
+     * Gets list of workspaces where given user is worker and has read permission
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/WorkspaceDao.java]: WorkspaceDao should check permissions?

Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/WorkspaceDao.java]: It should get all workspaces by relation between user and workspace. Take a look implementation in codenvy project https://github.com/codenvy/codenvy/blob/workspacePermissions/core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceDaoImpl.java#L264-264

**Comment by skabashnyuk** [comment]
ok

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java [hunk 2]
```diff
@@ -246,22 +246,22 @@ public WorkspaceImpl getWorkspace(String name, String namespace) throws NotFound
...
      *         when {@code owner} is null
      * @throws ServerException
-     *         when any server error occurs while getting workspaces with {@link WorkspaceDao#getByNamespace(String)}
+     *         when any server error occurs while getting workspaces with {@link WorkspaceDao#getWorkspaces(String)}
      */
-    public List<WorkspaceImpl> getWorkspaces(String namespace) throws ServerException {
-        requireNonNull(namespace, "Required non-null workspace namespace");
-        final List<WorkspaceImpl> workspaces = workspaceDao.getByNamespace(namespace);
+    public List<WorkspaceImpl> getWorkspaces(String user) throws ServerException {
+        requireNonNull(user, "Required non-null workspace namespace");
```
Comment (voievodin) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java]: fix namespace

Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java]: Fixed. Thx

**Comment by voievodin** [comment]
The `WorkspaceManager` looks ok.

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**